src/SqlPersistence/Outbox/OutboxPersister.cs

```diff
@@ -79,8 +81,10 @@ public async Task<OutboxMessage> Get(string messageId, ContextBag context)
                 command.CommandText = outboxCommands.Get;
                 command.Transaction = transaction;
                 command.AddParameter("MessageId", messageId);
-                // to avoid loading into memory SequentialAccess is required which means each fields needs to be accessed
-                using (var dataReader = await command.ExecuteReaderAsync(CommandBehavior.SingleRow | CommandBehavior.SequentialAccess).ConfigureAwait(false))
+
+                // to avoid loading into memory SequentialAccess is required which means each fields needs to be accessed, but SequentialAccess is unsupported for SQL Server AlwaysEncrypted
+                var behavior = sqlDialect.GetBehavior(connection);
```

Comment:
In line 79 the dialect is creating a command and then here we get the command behavior and apply it. I am wondering if we can't make it a bit simpler now that we have the `isEncrypted` in the dialect. I think we could change the `CommandWrapper` a bit:

```C#
public Task<DbDataReader> ExecuteReaderAsync(CommandBehavior behavior)
{
    var resultingBehavior = dialect.ModifyBehavior(behavior);
    return command.ExecuteReaderAsync(resultingBehavior);
}
```

and then change the `GetBehavior` into `ModifyBehavior`:

```C#
internal override CommandBehavior ModifyBehavior(CommandBehavior baseBehavior, DbConnection connection)
{
    if (!hasConnectionBeenInspectedForEncryption)
    {
        isConnectionEncrypted = connection.IsEncrypted();
        hasConnectionBeenInspectedForEncryption = true;
    }

    if (isConnectionEncrypted)
    {
        baseBehavior &= ~CommandBehavior.SequentialAccess; //Remove sequential access
    }

    return baseBehavior;
}
```
 
This would require passing the connection to the ctor of `CommandWrapper` but this should not be a big deal since it is available when the wrapper is created.

The benefit of this approach is that the code in the persisters would not need to be modified at all because it only passes the "base behaviors".